### PR TITLE
[BD-3379] Adding swift webview page

### PR DESCRIPTION
**Pull request by bovenkam**
### Summary

My PR adds a page that explains how to implement and use the Braze WebView for Swift

### Related PRs, issues, or features (optional)

- [BD-3379](https://jira.braze.com/browse/BD-3379)

### Feature release date (optional)

- Provides docs for a recently added simplification of our Swift WebView Bridge: (29/07-> https://github.com/braze-inc/braze-swift-sdk/releases/tag/10.0.0)

### Contributor checklist

- [x] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.

### CODE REVIEW

**Comment by bovenkam** [comment]
@lowip, can you double-check the code snippets? Thanks!

**Comment by internetisaiah** [comment]
@bovenkam , i got the deployment to build through [here](https://braze-docs-a9wg7kk3u-braze.vercel.app/docs/developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface). i'll start doing my content tweaks today 👍🏽

File: _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    </script>
+</head>
+<body>
+    <center>
+        <br><br>
+            <input type="button" value="Click to log a custom Event 'completed_level'" onclick="logData('completed_level')"><br>
+    </center>
+</body>
+</html>
```
Comment (lowip) [inline _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md:88]:
```suggestion
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Logging data via BrazeBridge Example</title>
    <script>
      function logData(data) {
        window.brazeBridge.logCustomEvent(data);
      }
    </script>
  </head>

  <body>
    <input
      type="button"
      value="Click to log a custom Event 'completed_level'"
      onclick="logData('completed_level')"
    />
  </body>
</html>
```

Formatted + removed noise.

File: _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+page_order: 5
+description: "This reference article shows how to add the Braze JavaScript Interface to WebViews."
+
+---
+
+# Braze JavaScript interface
+
+> This reference article shows how to add the Braze JavaScript Interface to iOS WebViews.
+
+Using Braze functionality from a WebView in your app can be done by adding the Braze JavaScript interface to your WebView. After the interface has been added, the same API available for [HTML in-app messages][1] will be available within your custom WebView.
```
Comment (lowip) [inline _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md:14]: Link is missing for _[HTML in-app messages][1]_.

**Review by lowip — APPROVED** [review]
LGTM after addressing comments

File: _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md (hunk 2 above)
Comment (bovenkam) [reply _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md:14]: Thanks, I've restored the link:

[1]: {{site.baseurl}}/user_guide/message_building_by_channel/in-app_messages/customize/#custom-html-messages

**Review by bovenkam — COMMENTED** [review]
Thanks, I've accepted the changes and added the IAM link.

File: _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+We'll start by adding the code required to use a WebView in our app
```
Comment (internetisaiah) [inline _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md:16]: > FROM DM:

I just want to clarify something as I'm working on your rewrites:

Is adding `ScriptMessageHandler` to the app, the same as adding `ScriptMessageHandler` to a WkWebView's `userContentController`?

Or are these two separate steps? If they are two separate steps, could you elaborate in this comment--then I'll update the doc!

Comment (internetisaiah) [reply _docs/_developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface.md:16]: > FROM DM:

They are different steps, we first have to add the sScriptMessageHandlers to our app, that's this code:

```
let scriptMessageHandler = Braze.WebViewBridge.ScriptMessageHandler(braze: braze)
```

Then customers add that `scriptMessageHandler` to a WkWebView's `userContentController`:

```
configuration.userContentController.add(
  scriptMessageHandler,
  name: Braze.WebViewBridge.ScriptMessageHandler.name
)
```

So in the first step we 'activate' the `ScriptMessageHandler` and pass through the braze instance and then in the second step we attach that newly created `scriptMessageHandler` to our webview.

**Comment by internetisaiah** [comment]
I just made some tweaks @bovenkam if you want to take a quick look before I merge. Let me know if I should tweak/change anything! Here's the latest deployment:

https://braze-docs-c2t65zrdl-braze.vercel.app/docs/developer_guide/platform_integration_guides/swift/advanced_use_cases/adding_braze_js_interface/

**Review by internetisaiah — APPROVED** [review]
LGTM

**Comment by bovenkam** [comment]
@internetisaiah Thanks, LGTM too!

**Pull request merged**